esp-metadata-generated/src/_generated_esp32.rs

```diff
@@ -1518,3 +1518,284 @@ macro_rules! define_io_mux_reg {
         }
     };
 }
+#[macro_export]
+#[doc(hidden)]
+macro_rules! implement_alternate_function_markers {
+    () => {
+        #[instability::unstable]
+        #[doc = "Marker trait for pins that have the SD_CMD input function"]
+        pub trait InputFunction_SD_CMD {}
```

Comment:
I understand wanting to keep the same naming convention for the TRM, but is it really worth having N copies of a marker trait for the same pin? `SD_CMD` is the same as `SDIO_CMD` on the esp32 and esp32c6 respectively.

What does the trait bounds on a `Sdio::new` look like? We'd have to import these seperately with cfg_if, then alias them to some common name anyway, right? Or am I missing something?

Reply:
I think Scott is worried about this.

```rust
import_sdio_signal_traits!();

impl<'d> Sdio<'d> {
  #[cfg(esp32)]
  fn new(peri: SDIO<'d>, cmd: impl AlternativeFunction_SD_CMD + 'd, data0: impl AlternativeFunction_SD_DATA0 + 'd, ...) { ... }

  #[cfg(esp32c6)]
  fn new(peri: SDIO<'d>, cmd: impl AlternativeFunction_SDIO_CMD + 'd, data0: impl AlternativeFunction_SDIO_DATA0 + 'd, ...) { ... }
}
```

Reply:
Well there is the obvious solution of doing nothing, ignoring compile-time safety here. That SDMMC throws another wrench into any sort of system here, doesn't really surprise me at this point.

I can see us providing two drivers/constructors there, and the two slots themselves as separate peris, but that's not great either.

Reply:
> I can see us providing two drivers/constructors there, and the two slots themselves as separate peris, but that's not great either.

This makes sense to me. It's only ever two slots atm